java/src/server/plugin/PluginLoader.java

```diff
@@ -1,27 +1,94 @@
 package server.plugin;
-
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
-public class PluginLoader implements IPluginLoader {
+import static java.util.Arrays.stream;
 
+public class PluginLoader implements IPluginLoader {
 
+    /**
+     * @inheritDoc
+     */
     @Override
-    public List<IPlugin> loadConfig(List<PluginConfig> configs) {
-        return null;
+    public List<IPlugin> loadConfig(List<PluginConfig> configs, File folder) {
+        List<String> neededPlugins = configs.stream()
+                .map(o -> o.getName().toLowerCase())
+                .collect(Collectors.toList());
+
+        return Arrays.stream(folder.listFiles())
+                .filter(x -> ".jar".equals(x.getName().substring(x.getName().length() - 4, x.getName().length())))
```

Comment:
I love the lambdas. However, there are functions specifically designed for splitting filenames, and it looks like this would probably crash and burn for a file named `abc` in that directory. I'm a little worried about the whole `toLowerCase` above - why is it there?